source/extensions/filters/http/json_to_metadata/filter.cc

```diff
@@ -189,42 +203,49 @@ void Filter::finalizeDynamicMetadata(Http::StreamFilterCallbacks& filter_callbac
   }
```

Comment:
we should do this in request_only given route is not changed after router filter in request.